falcon/hooks.py

```diff
@@ -109,7 +127,9 @@ def let(responder=responder):
     return _before
 
 
-def after(action, *args, is_async=False, **kwargs):
+def after(
+    action: Resource, *args: Any, is_async: bool = False, **kwargs: Any
```

Comment:
Similar comments to the above; also review code below and apply more precise type hints as required.

Reply:
Changed